pilot/pkg/model/push_context.go

```diff
@@ -1953,11 +1966,16 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) {
 		return in < jn
 	})
 
-	ps.envoyFiltersByNamespace = make(map[string][]*EnvoyFilterWrapper)
 	for _, envoyFilterConfig := range envoyFilterConfigs {
-		efw := convertToEnvoyFilterWrapper(&envoyFilterConfig)
-		if _, exists := ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace]; !exists {
-			ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace] = make([]*EnvoyFilterWrapper, 0)
+		key := ConfigKey{Kind: kind.EnvoyFilter, Namespace: envoyFilterConfig.Namespace, Name: envoyFilterConfig.Name}
+		var efw *EnvoyFilterWrapper
+		if changed.Contains(key) {
+			efw = convertToEnvoyFilterWrapper(&envoyFilterConfig)
+		} else if prev, ok := previous[key]; ok && features.OptimizedConfigRebuild {
+			// reuse the previous EnvoyFilterWrapper if it exists and optimized config rebuild is enabled
+			efw = prev
+		} else {
+			efw = convertToEnvoyFilterWrapper(&envoyFilterConfig)
```

Comment:
comment for this case it is new added one

Reply:
Isn't newly added one covered by L1972(part of changed)? Not sure when this condition will be hit

Reply:
yep, you are right.

This code can be run when `OptimizedConfigRebuild = false`

Reply:
it could happen if `OptimizedConfigRebuild == false` or if for some reason we get an element that wasn't previously there, this should probably never happen, but best to have it in place.

Reply:
I think it should never happen. But sure keeping it is safe. Can you add comments to that affect?

Reply:
done